integration_tests/steps/kpi.py

```diff
@@ -85,3 +85,22 @@ def penalty_rate_respected(context, device_name):
     assert isclose(bills[str(device.uuid)]["penalty_cost"],
                    bills[str(device.uuid)]["penalty_energy"] * DEVICE_PENALTY_RATE / 100.0,
                    rel_tol=0.0003 * DEVICE_PENALTY_RATE)
+
+
+@then('totals_with_penalties are correctly populated for all areas in the bills')
+def totals_with_penalties_bills_test(context):
+    bills = context.simulation.endpoint_buffer.market_bills.bills_results
+    expected_grid_totals = sum(bills[house]["totals_with_penalties"]["costs"]
```

Comment:
Name `expected` is a bit deceiving as it is extracting the calculated_value

Reply:
Actually no, it IS the expected value (the sum of all `totals_with_penalties` of all children.)
The actual value is `bills["Grid"]["totals_with_penalties"]["costs"]` (which i need to check)